medallion/test/test_memory_backend.py

```diff
@@ -372,3 +376,190 @@ def test_saving_data_file(self):  # just for the memory backend
+            headers=self.auth
+        )
+        self.assertEqual(r.status_code, 403)
+
+        # add_objects
+        new_id = "indicator--%s" % uuid.uuid4()
+        new_bundle = copy.deepcopy(API_OBJECTS_2)
+        new_bundle["objects"][0]["id"] = new_id
+
+        post_header = copy.deepcopy(self.auth)
+        post_header["Content-Type"] = MEDIA_TYPE_STIX_V20
+        post_header["Accept"] = MEDIA_TYPE_TAXII_V20
+
+        r_post = self.client.post(
+            "/trustgroup1/collections/64993447-4d7e-4f70-b94d-d7f33742ee63/objects/",
+            data=json.dumps(new_bundle),
+            headers=post_header
+        )
+        self.assertEqual(r_post.status_code, 403)
+
+    def test_get_or_add_objects_404(self):
+        # get_objects()
+        r = self.client.get(
+            "/trustgroup1/collections/12345678-1234-1234-1234-123456789012/objects/",
+            headers=self.auth,
+        )
+        self.assertEqual(r.status_code, 404)
+
+        # add_objects
+        new_id = "indicator--%s" % uuid.uuid4()
+        new_bundle = copy.deepcopy(API_OBJECTS_2)
+        new_bundle["objects"][0]["id"] = new_id
+
+        post_header = copy.deepcopy(self.auth)
+        post_header["Content-Type"] = MEDIA_TYPE_STIX_V20
+        post_header["Accept"] = MEDIA_TYPE_TAXII_V20
+
+        r_post = self.client.post(
+            "/trustgroup1/collections/12345678-1234-1234-1234-123456789012/objects/",
+            data=json.dumps(new_bundle),
+            headers=post_header
+        )
+        self.assertEqual(r_post.status_code, 404)
+
+    def test_get_or_add_objects_422(self):
+        """only applies to adding objects as would arise if user content is malformed"""
+
+        # add_objects()
+        new_id = "indicator--%s" % uuid.uuid4()
+        malformed_bundle = {
+            "created": "2016-11-03T12:30:59.000Z",
+            "description": "Accessing this url will infect your machine with malware.",
+            "id": new_id,
+            "labels": [
+                "url-watchlist"
+            ],
+            "modified": "2016-11-03T12:30:59.000Z",
+            "name": "Malicious site hosting downloader",
+            "pattern": "[url:value = 'http://yarg.cn/4712']",
+            "type": "indicator",
+            "valid_from": "2017-01-27T13:51:53.935382Z"
+        }
+
+        post_header = copy.deepcopy(self.auth)
+        post_header["Content-Type"] = MEDIA_TYPE_STIX_V20
+        post_header["Accept"] = MEDIA_TYPE_TAXII_V20
+
+        r_post = self.client.post(
+            "/trustgroup1/collections/91a7b528-80eb-42ed-a74d-c6fbd5a26116/objects/",
+            data=json.dumps(malformed_bundle),
+            headers=post_header
+        )
+        self.assertEqual(r_post.status_code, 422)
+        self.assertEqual(r_post.content_type, MEDIA_TYPE_TAXII_V20)
+        error_data = self.load_json_response(r_post.data)
+        assert error_data["title"] == "ProcessingError"
```

Comment:
I like that we're testing the various parameters on the response, and not just the HTTP status code 👍